core/platform/search/elastic_search_services.py

```diff
@@ -0,0 +1,167 @@
+
+    Raises:
+        Exception. Document id does not exist.
+    """
+    assert isinstance(index_name, python_utils.BASESTRING)
+    for doc_id in doc_ids:
+        assert isinstance(doc_id, python_utils.BASESTRING)
+
+    for doc_id in doc_ids:
+        if ES.exists(index_name, doc_id):
+            ES.delete(index_name, doc_id)
+        else:
+            raise Exception(
+                'Document id does not exist: %s' % doc_id)
+
+
+def clear_index(index_name):
+    """Clears an index on the elastic search instance.
+
+    Args:
+        index_name: str. The name of the index to clear.
+    """
+    assert isinstance(index_name, python_utils.BASESTRING)
+    # More details on clearing an index can be found here:
+    # https://elasticsearch-py.readthedocs.io/en/master/api.html#elasticsearch.Elasticsearch.delete_by_query
+    ES.delete_by_query(
+        index_name,
+        {
+            'query':
+                {
+                    'match_all': {}
+                }
+        })
+
+
+def search(
+        query_string, index_name, offset=None,
+        limit=feconf.SEARCH_RESULTS_PAGE_SIZE, ids_only=False):
+    """Searches for documents matching the given query in the given index.
+    NOTE: We cannot search through more than 10,000 pages from a search by
+    paginating using limit and offset. If the number of items to search through
+    is greater that 10,000, use the elasticsearch scroll API instead.
+
+    Args:
+        query_string: str. The search definition using Query DSL.
+        index_name: str. The name of the index. Use '_all' or empty string to
+            perform the operation on all indices.
+        offset: int. The offset into the index. Pass this in to start at the
+            'offset' when searching through a list of results of max length
+            'limit'. Leave as None to start at the beginning.
+        limit: int. The maximum number of documents to return.
+        ids_only: bool. Whether to only return document ids.
+
+    Returns:
+        2-tuple of (result_docs, resulting_offset). Where:
+            result_docs: list(dict). Represents search documents. If ids_only is
+                True, this will be a list of strings, doc_ids.
+            resulting_offset: int. The resulting offset to start at for the next
+                section of the resulting pages. Return None if there are no more
+                pages.
+    """
+    assert offset + limit < MAXIMUM_NUMBER_OF_PAGES
+    query_definiton = json.loads(query_string)
+    response = ES.search(
+        body=query_definiton, index=index_name,
+        params={
+            'size': limit,
+            'from': offset
+        })
+    resulting_offset = None
+    if len(response['hits']['hits']) != 0:
+        resulting_offset = offset + limit
+    if ids_only:
+        result_docs = [doc['_id'] for doc in response['hits']['hits']]
+    else:
+        result_docs = [doc['_source'] for doc in response['hits']['hits']]
```

Comment:
Just to check, does doc['_source'] include the id as well? If so, it might be worth mentioning this in a comment or in the docstring.